lisa/sut_orchestrator/baremetal/cluster/readychecker.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import os
+from dataclasses import dataclass, field
+from typing import Type
+
+from dataclasses_json import dataclass_json
+
+from lisa import schema
+from lisa.util import InitializableMixin, field_metadata, subclasses
+from lisa.util.logger import get_logger
+
+
+@dataclass_json()
+@dataclass
+class ReadyCheckerSchema(schema.TypedSchema, schema.ExtendableSchemaMixin):
+    type: str = field(default="file_single", metadata=field_metadata(required=True))
+
+
+class ReadyChecker(subclasses.BaseClassWithRunbookMixin, InitializableMixin):
+    def __init__(
+        self,
+        runbook: ReadyCheckerSchema,
+    ) -> None:
+        super().__init__(runbook=runbook)
+        self.ready_checker_runbook: ReadyCheckerSchema = self.runbook
+        self._log = get_logger("ready_checker", self.__class__.__name__)
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "ready_checker"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return ReadyCheckerSchema
+
+    def clean_up(self) -> None:
+        pass
+
+    def is_ready(self) -> bool:
+        pass
+
+
+@dataclass_json()
+@dataclass
+class FileSingleSchema(ReadyCheckerSchema):
+    file: str = ""
+
+
+class FileSingleChecker(ReadyChecker):
+    def __init__(
+        self,
+        runbook: FileSingleSchema,
+    ) -> None:
+        super().__init__(runbook=runbook)
+        self.file_single_runbook: FileSingleSchema = self.runbook
+        self._log = get_logger("file_single", self.__class__.__name__)
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "file_single"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return FileSingleSchema
+
+    def clean_up(self) -> None:
+        if os.path.exists(self.file_single_runbook.file):
+            os.remove(self.file_single_runbook.file)
+            self._log.debug(
+                f"The file {self.file_single_runbook.file} has been removed"
+            )
+        else:
+            self._log.debug(f"The file {self.file_single_runbook.file} does not exist")
```

Comment:
`"The file {self.file_single_runbook.file} does not exist, so it doesn't need to be cleaned up."`